src/stores/widgets/StopGapWidgetDriver.ts

```diff
@@ -485,4 +486,19 @@ export class StopGapWidgetDriver extends WidgetDriver {
             prevBatch: prevBatch ?? undefined,
         };
     }
+
+    public async searchUserDirectory(searchTerm: string, limit?: number): Promise<ISearchUserDirectoryResult> {
+        const client = MatrixClientPeg.get();
+
+        const { limited, results } = await client.searchUserDirectory({ term: searchTerm, limit });
```

Comment:
An alternative here could be to refactor and reuse the code in the [`InviteDialog`](https://github.com/matrix-org/matrix-react-sdk/blob/e5291c195d3259b5905d72dc2945ed740fffc6d1/src/components/views/dialogs/InviteDialog.tsx) to have the additional logic of making the search experience more personalised/useful. But I'm not sure if this is crucially important (see also https://github.com/matrix-org/matrix-spec-proposals/pull/3973#pullrequestreview-1320398823).

Reply:
As discussed in the MSC, we pass though the raw search results from the server. So there is no need for a more complex logic.